# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../mesa/build.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Use the compilers T2 is configured to, not only for cross builds, but
also for ICC, LLVM, etc. pp.

  - Rene Rebe <rene@exactcode.de>

And pick the right RANLIB when cross-compiling.

  - Michael Tross <michael@tross.org>

--- Mesa-7.0.3/bin/mklib.vanilla	2008-08-11 12:28:08.000000000 +0200
+++ Mesa-7.0.3/bin/mklib	2008-08-11 12:28:32.000000000 +0200
@@ -204,9 +204,9 @@
 	if [ "x$LINK" = "x" ] ; then
 	    # -linker was not specified so set default link command now
             if [ $CPLUSPLUS = 1 ] ; then
-                LINK=g++
+                LINK=$CXX
             else
-                LINK=gcc
+                LINK=$CC
             fi
 	fi
 
@@ -262,7 +262,7 @@
             rm -f ${LIBNAME}
             # make lib
             ${LINK} ${OPTS} ${LIBNAME} ${OBJECTS}
-            ranlib ${LIBNAME}
+            ${RANLIB} ${LIBNAME}
             # finish up
             FINAL_LIBS=${LIBNAME}
         else
